frontendExternalSystem/src/main/java/com/virtuslab/gitmachete/frontend/externalsystem/project/MacheteAutoImportAware.java

```diff
@@ -0,0 +1,29 @@
+package com.virtuslab.gitmachete.frontend.externalsystem.project;
+
+import java.io.File;
+import java.util.Objects;
+
+import com.intellij.openapi.components.Service;
+import com.intellij.openapi.externalSystem.ExternalSystemAutoImportAware;
+import com.intellij.openapi.project.Project;
+import git4idea.GitUtil;
+import io.vavr.collection.List;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+@Service
+public final class MacheteAutoImportAware implements ExternalSystemAutoImportAware {
+
+  @Override
+  public @Nullable String getAffectedExternalProjectPath(String changedFileOrDirPath, Project project) {
+    return null;
+  }
+
+  @Override
+  public java.util.List<File> getAffectedExternalProjectFiles(String projectPath, Project project) {
+    return List.ofAll(GitUtil.getRepositories(project))
+        .map(Objects::toString)
+        .map(path -> path.concat("/.git/machete"))
+        .map(File::new)
+        .toJavaList();
+  }
```

Comment:
I think that reling on `toString` is brittle.
Maybe something like this would be better:
```suggestion
  @Override
  @SuppressWarnings("nullness")
  public java.util.List<File> getAffectedExternalProjectFiles(String projectPath, Project project) {
    return List.ofAll(GitUtil.getRepositories(project))
        .map(repository -> GitUtil.findGitDir(repository.getRoot()))
        .filter(Objects::nonNull)
        .map(gitDirectory -> gitDirectory.findChild("machete"))
        .filter(Objects::nonNull)
        .map(VirtualFile::getPath)
        .map(File::new)
        .toJavaList();
  }
```

Reply:
Yeah, `toString` just happens to be `getPresentableUrl()` for the sole existing impl of `GitRepository` as of 2020.1... but a new implementing class can be added at some point and/or `toString` impl can change :/